src/Plugin/views/field/SortedRolesViewsField.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Drupal\tide_core\Plugin\views\field;
+
+use Drupal\views\Plugin\views\field\FieldPluginBase;
+use Drupal\views\ResultRow;
+use Drupal\views\Plugin\views\display\DisplayPluginBase;
+use Drupal\views\ViewExecutable;
+use Drupal\user\Entity\User;
+use Drupal\user\Entity\Role;
+
+/**
+ * A handler to provide a field that is completely custom by the administrator.
+ *
+ * @ingroup views_field_handlers
+ *
+ * @ViewsField("sorted_roles_views_field")
+ */
+class SortedRolesViewsField extends FieldPluginBase {
+  /**
+   * The current display.
+   *
+   * @var string
+   *   The current display of the view.
+   */
+  protected $currentDisplay;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function init(ViewExecutable $view, DisplayPluginBase $display, array &$options = NULL) {
+    parent::init($view, $display, $options);
+    $this->currentDisplay = $view->current_display;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function usesGroupBy() {
+    return FALSE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function query() {
+    // Do nothing -- to override the parent query.
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+    $options['hide_alter_empty'] = ['default' => FALSE];
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render(ResultRow $values) {
```

Comment:
by using overridding https://git.drupalcode.org/project/drupal/-/blob/10.1.x/core/modules/views/src/Plugin/views/field/PrerenderList.php
you don't need to consider cache issues, which you didn't add in your changes.

Reply:
@vincent-gao do you mean you _do_ need to consider cache issues? Can you give a little bit more explanation on this please?

Reply:
hey @anthony-malkoun / @yeniatencio 
Sorry for confusing you. I mean, in this case, by extending the base branch provided by the core module, we may not need to consider the cache because the parent call will take care of this. 
But it seems like its parent class doesn't cache the list, so maybe the context cache exists somewhere. 😬 .

Reply:
oh, the cache setting is in the View's advanced settings. seems like it's turned off.